haystack/nodes/file_converter/json.py

```diff
@@ -0,0 +1,102 @@
+    """Extracts text from JSON files and casts it into Document objects."""
+
+    outgoing_edges = 1
+
+    def convert(
+        self,
+        file_path: Path,
+        meta: Optional[Dict[str, Any]] = None,
+        remove_numeric_tables: Optional[bool] = None,
+        valid_languages: Optional[List[str]] = None,
+        encoding: Optional[str] = "UTF-8",
+        id_hash_keys: Optional[List[str]] = None,
+    ) -> List[Document]:
+        """
+        Reads a JSON file and converts it into a list of Documents.
+
+        It's a wrapper around `Document.from_dict()` and, as such, acts as the inverse of `Document.to_dict()`.
+
+        It expects one of these formats:
+        - A JSON file with a list of Document dicts.
+        - A JSONL file with every line containing either a Document dict or a list of dicts.
+
+        :param file_path: Path to the JSON file you want to convert.
+        :param meta: Optional dictionary with metadata you want to attach to all resulting documents.
+                     Can be any custom keys and values.
+                     The result will have a union of metadata specified here and already present in the json.
+                     In case of same keys being used, the one passed here takes precedence/overwrites the one from the json.
+        :param remove_numeric_tables: Uses heuristics to remove numeric rows from the tables.
+                                      The tabular structures in documents might be noise for the Reader model if it
+                                      can't parse tables to find the answers. But tables
+                                      may also have long strings that could be possible candidates for searching answers.
+                                      This option retains the rows containing strings.
+        :param valid_languages: Validates languages from a list of languages specified in the [ISO 639-1]                   (https://en.wikipedia.org/wiki/ISO_639-1) format.
+                                Use this option to test for encoding errors. If the extracted text is
+                                not one of the valid languages, it may be an encoding error resulting
+                                in garbled text.
+        :param encoding: Encoding that overwrites self.encoding and is passed as the `-enc` parameter to `pdftotext`.
+                         (Run `pdftotext -listenc` in the terminal to see the list of available encodings.)
+        :param id_hash_keys: Generate the document id from a custom list of strings that refer to the document's
+            attributes. To ensure you don't have duplicate documents in your DocumentStore if texts are
+            not unique, modify the metadata and pass, for example, `"meta"` to this field (example: [`"content"`, `"meta"`]).
+            The id is then generated by using the content and the defined metadata.
+            If specified it will overwrite any `id_hash_keys` present in the json file.
```

Comment:
I think I would expand on this docstring to include that if `id_hash_keys` is specified here or in the `__init__` of `JSONConverter` that it will overwrite the ones found in the json file.